Mathlib/Data/Multiset/Order.lean

```diff
@@ -0,0 +1,552 @@
+        have : ∃ a, a ∈ Y := by
+          rw [← Y.card_pos_iff_exists_mem]
+          cases foo : Multiset.card Y
+          tauto
+          simp
+        rcases this with ⟨y,claim⟩
+        let newY := Y.erase y
+        have newY_nonEmpty : newY ≠ ∅ := by
+          have : ∀ (n : ℕ), n ≠ 0 → n ≠ 1 → n ≥ 2 := by
+            intros n h0 h1
+            cases n
+            case zero => contradiction
+            case succ m =>
+              cases m
+              case zero => contradiction
+              case succ n=>
+                apply Nat.succ_le_succ
+                simp_all only [le_add_iff_nonneg_left, zero_le]
+          have : 0 < Multiset.card (Multiset.erase Y y) := by aesop
+          rw [Multiset.card_pos] at this
+          simp_all only [Multiset.empty_eq_zero, ne_eq, not_false_eq_true]
+        have newY_sub_Y : newY < Y := by simp (config := {zetaDelta := true}); exact claim
+        let f : α → Multiset α := fun y' => X.filter (fun x => x < y') -- DecidableRel
+        let N' := Z + newY + f y
+        apply @transitive_transGen _ _ _ N'
+        -- step from N' to M
+        · apply IH newY newY_sub_Y newY_nonEmpty
+          change M = (Z + f y) + (X - f y)
+          · ext a
+            have count_lt := Multiset.count_le_of_le a (Multiset.filter_le (fun x => x < y) X)
+            rw [M_def]
+            simp_all only [Multiset.empty_eq_zero, ne_eq, Multiset.card_eq_zero, not_false_eq_true,
+              Multiset.count_add, Multiset.count_sub]
+            let x := Multiset.count a X
+            let z := Multiset.count a Z
+            let fx := Multiset.count a (Multiset.filter (fun x => x < y) X)
+            change z + x = z + fx + (x - fx)
+            change fx ≤ x at count_lt
+            have : x = fx + (x - fx) := by simp_all only [add_tsub_cancel_of_le]
+            linarith
+          · unfold_let N'
+            rw [add_assoc, add_assoc, add_comm newY (f y)]
+          · intro x x_in
+            let X_lt_Y := X_lt_Y x
+            have x_in_X : x ∈ X := by
+              have Xfy_le_X : X - f y ≤ X:= by simp (config := {zetaDelta := true})
+              apply Multiset.mem_of_le Xfy_le_X
+              exact x_in
+            let X_lt_Y := X_lt_Y x_in_X
+            rcases X_lt_Y with ⟨t, t_in_Y, x_lt_t⟩
+            use t
+            constructor
+            · if t_in_newY : t ∈ newY then
+                exact t_in_newY
+                else
+                  exfalso
+                  have : t = y := by
+                    have : Y = newY + {y} := by
+                      unfold_let newY
+                      rw [add_comm, Multiset.singleton_add]
+                      simp [Multiset.cons_erase claim]
+                    rw [this] at t_in_Y
+                    rw [Multiset.mem_add] at t_in_Y
+                    have : t ∈ ( {y} : Multiset α) := by exact Or.resolve_left t_in_Y t_in_newY
+                    rw [← Multiset.mem_singleton]
+                    assumption
```

Comment:
```suggestion
                    rwa [← Multiset.mem_singleton]
```
You can replace `simp; assumption` and `rw [...]; assumption` by `simpa` and `rwa`.